tests/unit/NewConfigTests.cpp

```diff
@@ -0,0 +1,41 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2024, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include "util/newconfig/ConfigDefinition.hpp"
+
+#include <boost/json/object.hpp>
+#include <boost/json/parse.hpp>
+#include <boost/json/value.hpp>
+#include <boost/json/value_to.hpp>
+#include <gtest/gtest.h>
+
+using namespace util::config;
+
+// TODO: parsing config file and populating into config will be here once implemented
+
+TEST(ConfigDescription, getValues)
+{
+    ClioConfigDescription definition{
+        {{"database.type", "Type of database."}, {"key", "value"}, {"array[].sub", "this is the subsection."}}
+    };
+
+    ASSERT_EQ(definition.get("key"), "value");
```

Comment:
Here and in other tests: ASSERT vs. EXPECT is important to understand. Most of the time, you will want EXPECT_* instead of the ASSERT_* alternatives. ASSERT will immediately halt the test and error out while EXPECT will report the issue and continue trying. With EXPECT you may get more errors at once. With ASSERT you will likely only get the first one. Sometimes it makes sense to use ASSERT though - mostly when continuing with execution not be ok. Like when you check your vector to contain at least N elements with an ASSERT but later checks (within bounds) on the vector can be simple EXPECTED calls instead.